Renaming files fails between different filesystems

It seems `shutil.move` is struggling with cross-filesystems renames:

```
[worker] [2022-01-25 14:10:37,131] [ERROR] Task striparr.worker[8d8017b5-87ff-4e19-7271-084c26095fa9] raised unexpected: PermissionError(1, 'Operation not permitted')
[worker] Traceback (most recent call last):
[worker]   File "/usr/lib/python3.9/shutil.py", line 806, in move
[worker]     os.rename(src, real_dst)
[worker] OSError: [Errno 18] Invalid cross-device link: '/tmp/S12E03 - Foo.striparr.mkv' -> '/series/Foo/Season 12/S12E03 - Foo.mkv'
[worker]
[worker] During handling of the above exception, another exception occurred:
[worker]
[worker] Traceback (most recent call last):
[worker]   File "/usr/local/lib/python3.9/dist-packages/celery/app/trace.py", line 451, in trace_task
[worker]     R = retval = fun(*args, **kwargs)
[worker]   File "/usr/local/lib/python3.9/dist-packages/celery/app/trace.py", line 734, in __protected_call__
[worker]     return self.run(*args, **kwargs)
[worker]   File "/striparr.py", line 164, in worker
[worker]     shutil.move(stripped_filepath, filetostrip)
[worker]   File "/usr/lib/python3.9/shutil.py", line 820, in move
[worker]     copy_function(src, real_dst)
[worker]   File "/usr/lib/python3.9/shutil.py", line 436, in copy2
[worker]     copystat(src, dst, follow_symlinks=follow_symlinks)
[worker]   File "/usr/lib/python3.9/shutil.py", line 381, in copystat
[worker]     lookup("chmod")(dst, mode, follow_symlinks=follow)
[worker] PermissionError: [Errno 1] Operation not permitted: '/series/Foo/Season 12/S12E03 - Foo.mkv'
```

Potential solution: https://alexwlchan.net/2019/03/atomic-cross-filesystem-moves-in-python/

--- a/rootfs/striparr.py
+++ b/rootfs/striparr.py
@@ -161,7 +161,19 @@ def worker(filetostrip):
                 else:
                     # if output good, then overwrite original file
                     celery_logger.info('Overwriting: "%s" with clean file: "%s"' % (filetostrip, stripped_filepath))
-                    shutil.move(stripped_filepath, filetostrip)
+
+                    # attempt shutil move
+                    # if that fails, try copy/delete
+                    # see: https://alexwlchan.net/2019/03/atomic-cross-filesystem-moves-in-python/
+                    try:
+                        shutil.move(stripped_filepath, filetostrip)
+                    except OSError as err:
+                        if err.errno == errno.EXDEV:
+                            shutil.copyfile(stripped_filepath, filetostrip)
+                            os.unlink(stripped_filepath)
+                        else:
+                            raise
+                    
                     celery_logger.info('File is now clean: "%s"!' % (filetostrip))
 
         else:

--- a/rootfs/striparr.py
+++ b/rootfs/striparr.py
@@ -4,6 +4,7 @@ __version__ = "2021-01-09"
 
 import sys
 import os
+import errno
 import datetime
 import logging
 # import time

